### PR TITLE
Feat/extend background image

### LINKED ISSUE #37: Populate background image based on level size
## Description
- Depending on the size of the level, more background images should be rendered to fill the entire length of the level

### PR DESCRIPTION
**Pull request by jasonqiu212**
Resolves #37

### CODE REVIEW

**Review by ruihan00 — APPROVED** [review]
lgtm

File: star-dash/star-dash/ViewController.swift [hunk 1]
```diff
@@ -37,17 +40,24 @@ class ViewController: UIViewController {
         self.renderer = renderer
     }
 
-    func setupGameEntities() {
+    private func createGameEngine() -> GameEngine {
+        let levelSize = self.storageManager?.getLevelSize(id: 0) ?? CGSize(width: 4_842, height: 1_040)
```
Comment (Junyi00) [inline star-dash/star-dash/ViewController.swift:44]: I believe the id here should be consistent with method `setupGameEntities`, by passing a variable?

**Review by Junyi00 — CHANGES_REQUESTED** [review]
The overall implementation looks good and I appreciate the refactoring of the initialisation of GameEngine, GameScene and others. 

But I feel that there needs be a larger refactoring around the initialisation of the GameEngine itself. The loading of map entities, creation of player entities and creation of background image should probably not be the responsibility of the view controller... But I understand if this should be in another PR or if u would like to resolve it in this PR would be good too!

**Comment by ruihan00** [comment]
I agree that the game engine should handle the setting up of the entities, I will resolve this in the pr for Home Screen as it requires a refactoring of the game engine to set up the game

**Pull request merged**